internal/lsp/server.go

```diff
@@ -377,6 +384,32 @@ func (s *Server) handleDefinition(req *lsproto.RequestMessage) error {
 	return s.sendResult(req.ID, &lsproto.Definition{Locations: &lspLocations})
 }
 
+func (s *Server) handleCompletion(req *lsproto.RequestMessage) (messageErr error) {
+	params := req.Params.(*lsproto.CompletionParams)
+	file, project := s.getFileAndProject(params.TextDocument.Uri)
+	pos, err := s.converters.LineAndCharacterToPositionForFile(params.Position, file.FileName())
+	if err != nil {
+		return s.sendError(req.ID, err)
+	}
+
+	// !!! remove this after completions is fully ported/tested
+	defer func() {
+		if r := recover(); r != nil {
+			stack := debug.Stack()
+			fmt.Printf("panic obtaining completions: %v\n%s", r, string(stack))
```

Comment:
```suggestion
			s.Log("panic obtaining completions:", r, string(stack))
```

Otherwise, errors write to stdout and break the protocol.